pkg/discovery/probe/pod_probe.go

```diff
@@ -428,7 +428,7 @@ func (podProbe *PodProbe) buildPodEntityDTO(pod *api.Pod, commoditiesSold, commo
 	glog.V(4).Infof("Pod %s will be stitched with VM with %s: %s", podDisplayName, *property.Name, *property.Value)
 
 	// monitored or not
-	monitored := monitored(pod)
+	monitored := monitored(pod, notMonitoredNodes)
```

Comment:
monitored is a util method and notMonitoredNodes is a global map. I think we don't want it to access a global variable from inside the method. 
I've discussed with Cheuk and we can improve the overall implementation in the future.